Jetcaster/wear/src/main/java/com/example/jetcaster/ui/library/QueueScreen.kt

```diff
@@ -0,0 +1,199 @@
+import com.google.android.horologist.compose.layout.ScreenScaffold
+import com.google.android.horologist.compose.layout.rememberResponsiveColumnState
+import com.google.android.horologist.compose.material.AlertDialog
+import com.google.android.horologist.compose.material.Button
+import com.google.android.horologist.compose.material.Chip
+import com.google.android.horologist.images.base.util.rememberVectorPainter
+import com.google.android.horologist.images.coil.CoilPaintable
+import com.google.android.horologist.media.ui.screens.entity.DefaultEntityScreenHeader
+import com.google.android.horologist.media.ui.screens.entity.EntityScreen
+
+@Composable fun QueueScreen(
+    onChangeSpeedButtonClick: () -> Unit,
+    onPlayButtonClick: () -> Unit,
+    onEpisodeItemClick: (EpisodeToPodcast) -> Unit,
+    onErrorDialogCancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    queueViewModel: QueueViewModel = hiltViewModel()
+) {
+    val uiState by queueViewModel.uiState.collectAsStateWithLifecycle()
+
+    PodcastDetailsScreen(
+        viewState = uiState,
+        onChangeSpeedButtonClick = onChangeSpeedButtonClick,
+        onEpisodeItemClick = onEpisodeItemClick,
+        onPlayEpisode = queueViewModel::onPlayEpisode,
+        onErrorDialogCancelClick = onErrorDialogCancelClick,
+        onPlayButtonClick = onPlayButtonClick,
+        modifier = modifier,
+    )
+}
+
+@Composable
+fun PodcastDetailsScreen(
+    viewState: QueueScreenState,
+    onChangeSpeedButtonClick: () -> Unit,
+    onPlayButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    onEpisodeItemClick: (EpisodeToPodcast) -> Unit,
+    onPlayEpisode: (PlayerEpisode) -> Unit,
+    onErrorDialogCancelClick: () -> Unit
+) {
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = padding(
+            first = ScalingLazyColumnDefaults.ItemType.Text,
+            last = ScalingLazyColumnDefaults.ItemType.Chip
+        )
+    )
+    ScreenScaffold(
+        scrollState = columnState,
+        modifier = modifier
+    ) {
+        when (viewState) {
+            is QueueScreenState.Loaded -> {
+                EntityScreen(
+                    columnState = columnState,
+                    headerContent = {
+                        DefaultEntityScreenHeader(
+                            title = stringResource(R.string.queue)
+                        )
+                    },
+                    buttonsContent = {
+                        ButtonsContent(
+                            episodes = viewState.episodeList,
+                            onChangeSpeedButtonClick = onChangeSpeedButtonClick,
+                            onPlayButtonClick = onPlayButtonClick,
+                            onPlayEpisode = onPlayEpisode
+                        )
+                    },
+                    content = {
+                        items(count = viewState.episodeList.size) {
```

Comment:
remove size, and inline the episode to the lambda param.

Reply:
thanks